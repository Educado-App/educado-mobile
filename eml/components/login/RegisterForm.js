import React, { useEffect, useState } from 'react';
import { View } from 'react-native';
import { loginUser, registerUser } from '../../api/userApi';
import FormTextField from '../general/forms/FormTextField';
import FormButton from '../general/forms/FormButton';
import PasswordEye from '../general/forms/PasswordEye';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import ShowAlert from '../general/ShowAlert';
import FormFieldAlert from '../general/forms/FormFieldAlert';
import { removeEmojis, validatePasswordContainsLetter, validatePasswordLength, validateEmail, validateName } from '../general/Validation';
import Text from '../general/Text';
import errorSwitch from '../general/errorSwitch';
import { useNavigation } from '@react-navigation/native';
import DialogNotification from '../general/DialogNotification';
import { AlertNotificationRoot } from 'react-native-alert-notification';
import tailwindConfig from '../../tailwind.config';
import { setUserInfo, setJWT } from '../../services/StorageService';

/**
 * Component for registering a new account in the system, used in the register screen
 * @returns {React.Element} Component containing the form for registering a new user
 */

export default function RegisterForm() {

	const tailwindColors = tailwindConfig.theme.colors;

	const navigation = useNavigation();
	const [firstName, setFirstName] = useState('');
	const [lastName, setLastName] = useState('');
	const [email, setEmail] = useState('');
	const [password, setPassword] = useState('');
	const [confirmPassword, setConfirmPassword] = useState('');

	const [emailAlert, setEmailAlert] = useState('');
	const [nameAlert, setNameAlert] = useState('');
	const [isAllInputValid, setIsAllInputValid] = useState(false);
	const [confirmPasswordAlert, setConfirmPasswordAlert] = useState('');

	// State variable to track password visibility
	const [showPassword, setShowPassword] = useState(false);
	const [showConfirmPassword, setShowConfirmPassword] = useState(false);

	// Password Constraint variables
	const [passwordContainsLetter, setPasswordContainsLetter] = useState(false);
	const [passwordLengthValid, setPasswordLengthValid] = useState(false);

	useEffect(() => {
		// Clear input and alerts on first render
		setFirstName('');
		setLastName('');
		setEmail('');
		setPassword('');
		setConfirmPassword('');

		setNameAlert('');
		setEmailAlert('');
		setIsAllInputValid(false);
		setConfirmPasswordAlert('');
	}, []);

	useEffect(() => {
		const containsLetter = validatePasswordContainsLetter(password);
		setPasswordContainsLetter(containsLetter);
		const lengthValid = validatePasswordLength(password);
		setPasswordLengthValid(lengthValid);
		checkIfPasswordsMatch(password, confirmPassword);
	}, [password]);

	useEffect(() => {
		checkIfPasswordsMatch(password, confirmPassword);
	}, [confirmPassword]);

	useEffect(() => {
		let validationError = '';
		if (firstName !== '') {
			validationError = validateName(firstName, 'Nome'); // First name
		}
		if (validationError === '' && lastName !== '') {
			validationError = validateName(lastName, 'Sobrenome'); // Last name
		}

		setNameAlert(validationError);
	}, [firstName, lastName]);

	useEffect(() => {
		if (email === '') {
			setEmailAlert('');
			return;
		}

		const validationError = validateEmail(email);
		setEmailAlert(validationError);
	}, [email]);

	useEffect(() => {
		validateInput();
	}, [nameAlert, emailAlert, passwordLengthValid, passwordContainsLetter, confirmPasswordAlert]);

	// Functions to toggle password visibility states
	const toggleShowPassword = () => {
		setShowPassword(!showPassword);
	};

	const toggleShowConfirmPassword = () => {
		setShowConfirmPassword(!showConfirmPassword);
	};

	const checkIfPasswordsMatch = (password, confirmPassword) => {
		if (password === confirmPassword) {
			setConfirmPasswordAlert('');
		} else {
			// The passwords do not match
			setConfirmPasswordAlert('Os campos de senha precisam ser iguais');
		}
	};

	// TODO: This function should take into consideration
	// that alerts might be empty when input is yet to be given
	/**
   * Function for validating all input fields' content
   */
	function validateInput() {
		const validationPassed = (
			nameAlert === '' &&
      emailAlert === '' &&
      firstName != '' &&
      lastName != '' &&
      email != '' &&
      passwordLengthValid &&
      passwordContainsLetter &&
      confirmPasswordAlert === ''
		);

		setIsAllInputValid(validationPassed);
	}

	/**
   * Function for registering a new user in the database
   * @param {String} firstName 
   * @param {String} lastName
   * @param {String} email 
   * @param {String} password
   */
	async function register() {

		validateInput(firstName, email, password);

		if (!isAllInputValid) {
			return;
		}

		const obj = {
			firstName: firstName,
			lastName: lastName,
			email: email,
			password: password,
		};

		try {
			await registerUser(obj)
				.then(async function (response) {
					// Save user info in storage
					// TODO: Refactor backend to get the same response as on login
					const userInfo = {
						id: response.baseUser._id,
						...response.baseUser,
					};
					await setUserInfo(userInfo);
				}).then(async function () {
					// logs in the user, if no errors occur, navigates to home screen and sets token
					await loginFromRegister(obj);
				})
				.catch((error) => {
					ShowAlert(errorSwitch(error));
				});
		} catch (e) {
			console.log(e);
		}
	}

	/**
 * function to log in the user and set the login token, meant to be called after registering
 * @param {Object} obj the object containing the following fields:
 *  firstName: String
 *  lastName: String
 *  email: String
 */
	async function loginFromRegister(obj) {
		try {
			await loginUser(obj).then((response) => {
				setJWT(response.accessToken);
				DialogNotification('success', 'Usuário cadastrado! Cantando em...');
				setTimeout(() => {
					navigation.navigate('HomeStack');
				}, 2500);
			}).catch((error) => {
				console.log(error);
			});
		} catch (e) {
			console.log(e);
		}
	}


	return (
		<View>
			<AlertNotificationRoot>
				<View className="mb-6">
					<FormTextField
						label="Nome" // first name
						name={'Nome'}
						value={firstName}
						testId="firstNameInput"
						placeholder="Nome"
						required={true}
						onChangeText={(firstName) => {
							setFirstName(firstName);
						}}
					/>
				</View>
				<View className="mb-6">
					<FormTextField
						label="Sobrenome" // Last name
						name={'Sobrenome'}
						value={lastName}
						testId="lastNameInput"
						placeholder="Sobrenome"

						required={true}
						onChangeText={(lastName) => {
							setLastName(lastName);
						}}
					/>
					<FormFieldAlert label={nameAlert} />
				</View>
				<View className="mb-6">
					<FormTextField
						className="mb-6"
						label="E-mail"
						name={'E-mail'}
						testId="emailInput"
						value={email}
						placeholder="Insira sua e-mail"
						keyboardType="email-address"
						required={true}
						onChangeText={async (email) => { setEmail(email); validateEmail(email); }}
					/>
					<FormFieldAlert label={emailAlert} testId="emailAlert" />
				</View>
				<View className="mb-6">
					<View className="relative">
						<FormTextField
							label="Senha" //Password
							name={'Senha'}
							testId="passwordInput"
							value={password}
							placeholder="Insira sua senha" // Enter your password
							placeholderTextColor={tailwindColors.projectGray}
							secureTextEntry={!showPassword}
							required={true}
							onChangeText={(inputPassword) => {
								setPassword(removeEmojis(inputPassword, password));

							}}
						/>
						<PasswordEye
							testId="passwordEye"
							showPasswordIcon={showPassword}
							toggleShowPassword={toggleShowPassword}
						/>
					</View>

					<View className="flex-row justify-start mt-1 h-6">
						<Text testId="passwordLengthAlert" className={'text-xs' + ((passwordLengthValid || !password) ? ' text-projectGray' : ' text-error')}>
							{/* Minimum 8 characters */}
              • Mínimo 8 caracteres
						</Text>
						<View className="flex-row items-center -translate-y-1">
							{passwordLengthValid ? (
								<MaterialCommunityIcons name="check" size={20} color={tailwindColors.success} />
							) : null}
						</View>
					</View>
					<View className="flex-row justify-start h-6">
						<Text testId="passwordLetterAlert" className={'text-xs font-sans' + ((passwordContainsLetter || !password) ? ' text-projectGray' : ' text-error')}>
							{/* Must contain at least one letter */}
              • Conter pelo menos uma letra
						</Text>
						<View className="flex-row items-center -translate-y-1">
							{passwordContainsLetter ? (
								<MaterialCommunityIcons name="check" size={20} color={tailwindColors.success} />
							) : null}
						</View>
					</View>
				</View>
				<View className="mb-2">
					<View className="relative">
						<FormTextField
							label="Confirmar senha" // Confirm password
							value={confirmPassword}
							testId="confirmPasswordInput"
							onChangeText={(inputConfirmPassword) => {
								setConfirmPassword(removeEmojis(inputConfirmPassword, confirmPassword));
							}}

							placeholder="Confirme sua senha" // Confirm your password
							secureTextEntry={!showConfirmPassword}
							required={true}
						/>
						<PasswordEye
							testId="confirmPasswordEye"
							showPasswordIcon={showConfirmPassword}
							toggleShowPassword={toggleShowConfirmPassword}
						/>
					</View>
					<FormFieldAlert label={confirmPasswordAlert} />
				</View>
				{/* Register */}
				<View className="my-2">
					<FormButton
						onPress={() => register(firstName, lastName, email, password)}
						testId="registerButton"
						disabled={!isAllInputValid}
					>
            Cadastrar
					</FormButton>
				</View>
			</AlertNotificationRoot>
		</View>
	);
}
