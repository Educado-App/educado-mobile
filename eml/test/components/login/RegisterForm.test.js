import renderer from 'react-test-renderer';
import React from 'react';
import RegisterForm from '../../../components/login/RegisterForm';

let registerForm;

jest.useRealTimers('legacy');

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: jest.fn(),
  }),
}));

beforeEach(async () => {
  await renderer.act(() => {
    registerForm = renderer.create(<RegisterForm />);
  });
});

test('Ensure that the RegisterForm component renders correctly', () => {
  const tree = registerForm.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Check that password visibility is toggled correctly', async () => {
  const passwordEye = registerForm.root.findByProps({ testId: 'passwordEye' });
  expect(passwordEye.props.showPasswordIcon).toBe(false);
  await renderer.act(() => {
    passwordEye.props.toggleShowPassword();
  });
  expect(passwordEye.props.showPasswordIcon).toBe(true);
  await renderer.act(() => {
    passwordEye.props.toggleShowPassword();
  });
  expect(passwordEye.props.showPasswordIcon).toBe(false);
});

test('Test of email validation', async () => {
  const emailInput = registerForm.root.findByProps({
    testId: 'emailInput'
  });
  const emailAlert = registerForm.root.findByProps({
    testId: 'emailAlert'
  });

  await renderer.act(() => {
    emailInput.props.onChangeText('letters@letters.end').then(() => {
      expect(emailAlert.props.label)
        .toBe('');
    });
    emailInput.props.onChangeText('user.name@dev.letters.end').then(() => {
      expect(emailAlert.props.label)
        .toBe('');
    });
    emailInput.props.onChangeText('user-name@dev.letters.end').then(() => {
      expect(emailAlert.props.label)
        .toBe('');
    });
    emailInput.props.onChangeText('unusual----.++email+adress@dev.letters.end').then(() => {
      expect(emailAlert.props.label)
        .toBe('');
    });
  });

  await renderer.act(() => {
    emailInput.props.onChangeText('invalid email').then(() => { // No @, ., and illegal spaces
      expect(emailAlert.props.label)
        .toBe(not(''));
    });
    emailInput.props.onChangeText('slightly@invalid.d').then(() => { // Domain too short
      expect(emailAlert.props.label)
        .toBe(not(''));
    });
    emailInput.props.onChangeText('also.invalid.com').then(() => { // No @
      expect(emailAlert.props.label)
        .toBe(not(''));
    });
    emailInput.props.onChangeText('').then(() => { // Empty
      expect(emailAlert.props.label)
        .toBe(not(''));
    });
    emailInput.props.onChangeText('user.name.@domain.dom').then(() => { // Dot at end of local part
      expect(emailAlert.props.label)
        .toBe(not(''));
    });
    emailInput.props.onChangeText('username@domain-name.com').then(() => { // Dash in domain name
      expect(emailAlert.props.label)
        .toBe(not(''));
    });
    emailInput.props.onChangeText('username@domain?name.com').then(() => { // ? in domain name
      expect(emailAlert.props.label)
        .toBe(not(''));
    });
    emailInput.props.onChangeText('@domain.com').then(() => { // No local part
      expect(emailAlert.props.label)
        .toBe(not(''));
    });
    emailInput.props.onChangeText('name@email').then(() => { // No domain ending
      expect(emailAlert.props.label)
        .toBe(not(''));
    });
  });
});

test('Check that confirm password visibility is toggled correctly', async () => {
  expect(registerForm.root.findByProps({ testId: 'confirmPasswordEye' })
    .props.showPasswordIcon).toBe(false);
  await renderer.act(() => {
    registerForm.root.findByProps({ testId: 'confirmPasswordEye' })
      .props.toggleShowPassword();
  });
  expect(registerForm.root.findByProps({ testId: 'confirmPasswordEye' })
    .props.showPasswordIcon).toBe(true);
  await renderer.act(() => {
    registerForm.root.findByProps({ testId: 'confirmPasswordEye' })
      .props.toggleShowPassword();
  });
  expect(registerForm.root.findByProps({ testId: 'confirmPasswordEye' })
    .props.showPasswordIcon).toBe(false);
});

describe('password validation', () => {
  registerForm = renderer.create(<RegisterForm />);
  const passwordInput = registerForm.root.findByProps({
    testId: 'passwordInput'
  });
  const passwordLengthAlert = registerForm.root.findByProps({
    testId: 'passwordLengthAlert'
  });

  const passwordLetterAlert = registerForm.root.findByProps({
    testId: 'passwordLetterAlert'
  });

  const changePassword = async (password) => {
    await renderer.act(() => {
      passwordInput.props.onChangeText(password);
    });
  };

  const expectClass = (element, className) => {
    expect(element.props.className).toBe(className);
  };

  test('Test password length validation', async () => {
    await renderer.act(() => {
      changePassword('12345678a').then(() => {
        expectClass(passwordLengthAlert, 'text-xs font-montserrat text-gray');
      });
    });

    await renderer.act(() => {
      changePassword('1278a').then(() => {
        expectClass(passwordLengthAlert, 'text-xs font-montserrat text-error');
      });
    });
  });

  test('Test password letter inclusion validation', async () => {
    await renderer.act(() => {
      changePassword('testPassword').then(() => {
        expectClass(passwordLetterAlert, 'text-xs font-montserrat text-gray');
      });
    });

    await renderer.act(() => {
      changePassword('12783290189').then(() => {
        expectClass(passwordLengthAlert, 'text-xs font-montserrat text-error');
      });
    });
  });
});

test('Test that the password input works correctly', async () => {
  const passwordInput = registerForm.root.findByProps({
    testId: 'passwordInput'
  });
  renderer.act(() => {
    passwordInput.props.onChangeText('12345678a');
  }).then(() => {
    expect(passwordInput.props.value)
      .toBe('12345678a');
  });
});

test('Check that register button is disabled when fields are empty', async () => {
  const firstNameInput = registerForm.root.findByProps({ testId: 'firstNameInput' });
  const lastNameInput = registerForm.root.findByProps({ testId: 'lastNameInput' });
  const emailInput = registerForm.root.findByProps({ testId: 'emailInput' });
  const passwordInput = registerForm.root.findByProps({ testId: 'passwordInput' });
  const confirmPasswordInput = registerForm.root.findByProps({ testId: 'confirmPasswordInput' });
  const registerButton = registerForm.root.findByProps({ testId: 'registerButton' });

  // Check that register button is disabled when all fields are empty
  await renderer.act(() => {
    firstNameInput.props.onChangeText('');
    lastNameInput.props.onChangeText('');
    emailInput.props.onChangeText('');
    passwordInput.props.onChangeText('');
    confirmPasswordInput.props.onChangeText('');
  });
  expect(registerButton.props.disabled).toBe(true);
});


function testRegisterButtonDisable() {
  const firstNameInput = registerForm.root.findByProps({ testId: 'firstNameInput' });
  const lastNameInput = registerForm.root.findByProps({ testId: 'lastNameInput' });
  const emailInput = registerForm.root.findByProps({ testId: 'emailInput' });
  const passwordInput = registerForm.root.findByProps({ testId: 'passwordInput' });
  const confirmPasswordInput = registerForm.root.findByProps({ testId: 'confirmPasswordInput' });
  const registerButton = registerForm.root.findByProps({ testId: 'registerButton' });

  test('Check that register button is not disabled when fields are filled correctly', async () => {
    // Check that the button is disabled when only the name field is filled
    await renderer.act(async () => {
      firstNameInput.props.onChangeText('testFirstName');
      lastNameInput.props.onChangeText('testLastName');
      emailInput.props.onChangeText('test@test.dk');
      await passwordInput.props.onChangeText('passwordTest');
      await confirmPasswordInput.props.onChangeText('passwordTest');
    });
    expect(registerButton.props.disabled).toBe(false);
  });

  test('Check that register button is disabled when password and confirm password is not the same', async () => {
    // Test that the register button is disabled when the password and confirm password fields are not equal
    await renderer.act(() => {
      firstNameInput.props.onChangeText('firstname');
      lastNameInput.props.onChangeText('lastname');
      emailInput.props.onChangeText('test@test.dk');
      passwordInput.props.onChangeText('testing123');
      confirmPasswordInput.props.onChangeText('not the same password');
    });
    expect(registerButton.props.disabled).toBe(true);
  });
}

test('Password field filters out emojis', async () => {
  const passwordInput = registerForm.root.findByProps({ testId: 'passwordInput' });
  const confirmPasswordInput = registerForm.root.findByProps({ testId: 'confirmPasswordInput' });

  // reset text value before test
  await renderer.act(() => {
    passwordInput.props.onChangeText('');
    confirmPasswordInput.props.onChangeText('');
  });

  await renderer.act(() => {
    passwordInput.props.onChangeText('testing123');
    confirmPasswordInput.props.onChangeText('testing123');
  });
  expect(passwordInput.props.value).toBe('testing123');
  expect(confirmPasswordInput.props.value).toBe('testing123');

  await renderer.act(() => {
    passwordInput.props.onChangeText('testing123ðŸ¤”');
    confirmPasswordInput.props.onChangeText('testing123ðŸ¤”');
  });
  expect(passwordInput.props.value).toBe('testing123');
  expect(confirmPasswordInput.props.value).toBe('testing123');
});

testRegisterButtonDisable();