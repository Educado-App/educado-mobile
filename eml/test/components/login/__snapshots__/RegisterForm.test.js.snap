// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ensure that the RegisterForm component renders correctly 1`] = `
<View>
  <View
    style={
      [
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    <View>
      <View
        style={
          [
            {
              "display": "flex",
            },
            {
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              {
                "color": "#383838",
              },
              {
                "fontSize": 16,
              },
              [
                {
                  "marginLeft": 8,
                },
                {
                  "fontSize": 12,
                  "lineHeight": 16,
                },
              ],
            ]
          }
        >
          Nome
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              {
                "color": "#383838",
              },
              {
                "fontSize": 16,
              },
              [
                {
                  "marginLeft": 4,
                },
                {
                  "fontSize": 12,
                  "lineHeight": 16,
                },
                {
                  "color": "#FF4949",
                },
              ],
            ]
          }
        >
          *
        </Text>
      </View>
      <View>
        <TextInput
          autoComplete="off"
          keyboardType="default"
          onChangeText={[Function]}
          passwordGuidelines={false}
          placeholder="Nome"
          secureTextEntry={false}
          style={
            [
              {
                "paddingBottom": 4,
                "paddingTop": 4,
              },
              {
                "paddingLeft": 10,
              },
              {
                "backgroundColor": "#FFFFFF",
              },
              {
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
              },
            ]
          }
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    <View>
      <View
        style={
          [
            {
              "display": "flex",
            },
            {
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              {
                "color": "#383838",
              },
              {
                "fontSize": 16,
              },
              [
                {
                  "marginLeft": 8,
                },
                {
                  "fontSize": 12,
                  "lineHeight": 16,
                },
              ],
            ]
          }
        >
          Sobrenome
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              {
                "color": "#383838",
              },
              {
                "fontSize": 16,
              },
              [
                {
                  "marginLeft": 4,
                },
                {
                  "fontSize": 12,
                  "lineHeight": 16,
                },
                {
                  "color": "#FF4949",
                },
              ],
            ]
          }
        >
          *
        </Text>
      </View>
      <View>
        <TextInput
          autoComplete="off"
          keyboardType="default"
          onChangeText={[Function]}
          passwordGuidelines={false}
          placeholder="Sobrenome"
          secureTextEntry={false}
          style={
            [
              {
                "paddingBottom": 4,
                "paddingTop": 4,
              },
              {
                "paddingLeft": 10,
              },
              {
                "backgroundColor": "#FFFFFF",
              },
              {
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
              },
            ]
          }
          value=""
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "Montserrat-Regular",
            },
            {
              "color": "#383838",
            },
            {
              "fontSize": 16,
            },
            [
              {
                "fontSize": 12,
                "lineHeight": 16,
              },
              {
                "color": "#FF4949",
              },
              {
                "marginLeft": 8,
                "marginRight": 8,
              },
            ],
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      [
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    <View>
      <View
        style={
          [
            {
              "display": "flex",
            },
            {
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              {
                "color": "#383838",
              },
              {
                "fontSize": 16,
              },
              [
                {
                  "marginLeft": 8,
                },
                {
                  "fontSize": 12,
                  "lineHeight": 16,
                },
              ],
            ]
          }
        >
          E-mail
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              {
                "color": "#383838",
              },
              {
                "fontSize": 16,
              },
              [
                {
                  "marginLeft": 4,
                },
                {
                  "fontSize": 12,
                  "lineHeight": 16,
                },
                {
                  "color": "#FF4949",
                },
              ],
            ]
          }
        >
          *
        </Text>
      </View>
      <View>
        <TextInput
          autoComplete="off"
          keyboardType="email-address"
          onChangeText={[Function]}
          passwordGuidelines={false}
          placeholder="Insira sua e-mail"
          secureTextEntry={false}
          style={
            [
              {
                "paddingBottom": 4,
                "paddingTop": 4,
              },
              {
                "paddingLeft": 10,
              },
              {
                "backgroundColor": "#FFFFFF",
              },
              {
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
              },
            ]
          }
          value=""
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "Montserrat-Regular",
            },
            {
              "color": "#383838",
            },
            {
              "fontSize": 16,
            },
            [
              {
                "fontSize": 12,
                "lineHeight": 16,
              },
              {
                "color": "#FF4949",
              },
              {
                "marginLeft": 8,
                "marginRight": 8,
              },
            ],
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      [
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "position": "relative",
          },
        ]
      }
    >
      <View>
        <View
          style={
            [
              {
                "display": "flex",
              },
              {
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Montserrat-Regular",
                },
                {
                  "color": "#383838",
                },
                {
                  "fontSize": 16,
                },
                [
                  {
                    "marginLeft": 8,
                  },
                  {
                    "fontSize": 12,
                    "lineHeight": 16,
                  },
                ],
              ]
            }
          >
            Senha
          </Text>
          <Text
            style={
              [
                {
                  "fontFamily": "Montserrat-Regular",
                },
                {
                  "color": "#383838",
                },
                {
                  "fontSize": 16,
                },
                [
                  {
                    "marginLeft": 4,
                  },
                  {
                    "fontSize": 12,
                    "lineHeight": 16,
                  },
                  {
                    "color": "#FF4949",
                  },
                ],
              ]
            }
          >
            *
          </Text>
        </View>
        <View>
          <TextInput
            autoComplete="off"
            keyboardType="default"
            onChangeText={[Function]}
            passwordGuidelines={false}
            placeholder="Insira sua senha"
            secureTextEntry={true}
            style={
              [
                {
                  "paddingBottom": 4,
                  "paddingTop": 4,
                },
                {
                  "paddingLeft": 10,
                },
                {
                  "backgroundColor": "#FFFFFF",
                },
                {
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                },
              ]
            }
            value=""
          />
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "position": "absolute",
            },
            {
              "right": 0,
            },
            {
              "top": 8,
            },
            {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "gray",
                "fontSize": 24,
              },
              undefined,
              {
                "fontFamily": "material-community",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          󰈈
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "justifyContent": "flex-start",
          },
          {
            "marginTop": 4,
          },
          {
            "height": 24,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "Montserrat-Regular",
            },
            {
              "color": "#383838",
            },
            {
              "fontSize": 16,
            },
            [
              {
                "fontSize": 12,
                "lineHeight": 16,
              },
              {
                "color": "#A1ACB2",
              },
            ],
          ]
        }
        testId="passwordLengthAlert"
      >
        • Mínimo 8 caracteres
      </Text>
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "alignItems": "center",
            },
            {
              "transform": [
                {
                  "translateY": -4,
                },
                {
                  "translateX": 0,
                },
              ],
            },
          ]
        }
      />
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "justifyContent": "flex-start",
          },
          {
            "height": 24,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "Montserrat-Regular",
            },
            {
              "color": "#383838",
            },
            {
              "fontSize": 16,
            },
            [
              {
                "fontSize": 12,
                "lineHeight": 16,
              },
              {
                "fontFamily": "Montserrat-Regular",
              },
              {
                "color": "#A1ACB2",
              },
            ],
          ]
        }
        testId="passwordLetterAlert"
      >
        • Conter pelo menos uma letra
      </Text>
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "alignItems": "center",
            },
            {
              "transform": [
                {
                  "translateY": -4,
                },
                {
                  "translateX": 0,
                },
              ],
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      [
        {
          "marginBottom": 8,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "position": "relative",
          },
        ]
      }
    >
      <View>
        <View
          style={
            [
              {
                "display": "flex",
              },
              {
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Montserrat-Regular",
                },
                {
                  "color": "#383838",
                },
                {
                  "fontSize": 16,
                },
                [
                  {
                    "marginLeft": 8,
                  },
                  {
                    "fontSize": 12,
                    "lineHeight": 16,
                  },
                ],
              ]
            }
          >
            Confirmar senha
          </Text>
          <Text
            style={
              [
                {
                  "fontFamily": "Montserrat-Regular",
                },
                {
                  "color": "#383838",
                },
                {
                  "fontSize": 16,
                },
                [
                  {
                    "marginLeft": 4,
                  },
                  {
                    "fontSize": 12,
                    "lineHeight": 16,
                  },
                  {
                    "color": "#FF4949",
                  },
                ],
              ]
            }
          >
            *
          </Text>
        </View>
        <View>
          <TextInput
            autoComplete="off"
            keyboardType="default"
            onChangeText={[Function]}
            passwordGuidelines={false}
            placeholder="Confirme sua senha"
            secureTextEntry={true}
            style={
              [
                {
                  "paddingBottom": 4,
                  "paddingTop": 4,
                },
                {
                  "paddingLeft": 10,
                },
                {
                  "backgroundColor": "#FFFFFF",
                },
                {
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                },
              ]
            }
            value=""
          />
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "position": "absolute",
            },
            {
              "right": 0,
            },
            {
              "top": 8,
            },
            {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "gray",
                "fontSize": 24,
              },
              undefined,
              {
                "fontFamily": "material-community",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          󰈈
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "Montserrat-Regular",
            },
            {
              "color": "#383838",
            },
            {
              "fontSize": 16,
            },
            [
              {
                "fontSize": 12,
                "lineHeight": 16,
              },
              {
                "color": "#FF4949",
              },
              {
                "marginLeft": 8,
                "marginRight": 8,
              },
            ],
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      [
        {
          "marginBottom": 8,
          "marginTop": 8,
        },
      ]
    }
  >
    <View>
      <View
        accessibilityState={
          {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#5ECCE9",
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "opacity": 0.5,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              {
                "color": "#383838",
              },
              {
                "fontSize": 16,
              },
              [
                {
                  "textAlign": "center",
                },
                {
                  "fontFamily": "Montserrat-Bold",
                },
                {
                  "fontSize": 16,
                },
                {
                  "color": "#FFFFFF",
                },
              ],
            ]
          }
        >
          Cadastrar
        </Text>
      </View>
    </View>
  </View>
</View>
`;
